src/SAML2/XML/samlp/StatusDetail.php

```diff
@@ -18,41 +20,73 @@
 
 class StatusDetail extends \SAML2\XML\AbstractConvertable
 {
-    /** @var \DOMElement|null */
-    private $detail = null;
+    /** @var \SAML2\XML\Chunk[]|null */
+    private $details = null;
 
 
     /**
      * Initialize a samlp:StatusDetail
      *
-     * @param \DOMElement $detail
+     * @param \SAML2\XML\Chunk[] $details
      */
-    public function __construct(\DOMElement $detail = null)
+    public function __construct(array $details = null)
     {
-        $this->detail = $detail;
+        $this->setDetails($details);
     }
 
 
     /**
-     * Collect the detail
+     * Collect the details
      *
-     * @return \DOMElement|null
+     * @return \SAML2\XML\Chunk[]|null
      */
-    public function getDetail(): ?DOMElement
+    public function getDetails(): ?array
     {
-        return $this->detail;
+        return $this->details;
     }
 
 
     /**
-     * Set the value of the detail-property
+     * Set the value of the details-property
      *
-     * @param \DOMElement|null $detail
+     * @param \SAML2\XML\Chunk[]|null $details
      * @return void
      */
-    public function setDetail(?\DOMElement $detail): void
+    private function setDetails(?array $details): void
     {
-        $this->detail = $detail;
+        Assert::allIsInstanceOf($details, Chunk::class);
+        $this->details = $details;
+    }
+
+
+    /**
+     * Serialize this XML chunk
+     *
+     * @return string The serialized chunk.
+     */
+    public function serialize(): string
```

Comment:
`serialize()` and `unserialize()` shouldn't be necessary here either. `fromXML()` should handle reconstruction of the object.

Reply:
Removed